<launch>
	<!-- Launches the AR.Drone driver -->
	<node name="ardrone_driver" pkg="ardrone_autonomy" type="ardrone_driver" output="screen" clear_params="true" ns="alpha" args="-i 192.168.3.9 -p 15">
	    <param name="outdoor" value="1" />				<!-- If we are flying outdoors, will select more aggressive default settings -->
	    <param name="flight_without_shell" value="1" /> <!-- Changes internal controller gains if we are flying without the propeller guard -->
	    
	    <param name="altitude_max" value="6000" />		<!-- in millimeters = 3 meters = 9' -->
        <param name="altitude_min" value="300" />		<!-- in millimeters = 5cm = 2" -->
        <param name="euler_angle_max" value="0.35" />	<!-- maximum allowable body angle in radians = 5 degrees -->
        <param name="control_vz_max" value="2000" />		<!-- maximum z velocity in mm/sec = 0.2m/sec -->
	    <param name="control_yaw" value="6.11" />		<!-- maximum rotation rate in radians/sec = 40 degrees per second (1/9 rev/sec) -->

	    <param name="detect_type" value="10" />
	    <param name="detections_select_h" value="32" />	<!-- In the horizontal camera, look for the orange-blue-orange markers -->
	    <param name="detections_select_v_hsync" value="128" />  <!-- In the vertical camera, detect the A4 marker -->
	    <param name="enemy_colors" value="3" />			<!-- Orange-Blue-Orange = 3 -->
	    <param name="enemy_without_shell" value="0" />	<!-- Detect the indoor stickers, rather than the outdoor hull -->
	</node>

	<node name="ardrone_driver" pkg="ardrone_autonomy" type="ardrone_driver" output="screen" clear_params="true" ns="charlie" args="-i 192.168.3.11 -p 35">
	    <param name="outdoor" value="1" />				<!-- If we are flying outdoors, will select more aggressive default settings -->
	    <param name="flight_without_shell" value="1" /> <!-- Changes internal controller gains if we are flying without the propeller guard -->
	    
	    <param name="altitude_max" value="6000" />		<!-- in millimeters = 3 meters = 9' -->
        <param name="altitude_min" value="50" />		<!-- in millimeters = 5cm = 2" -->
        <param name="euler_angle_max" value="0.35" />	<!-- maximum allowable body angle in radians = 5 degrees -->
        <param name="control_vz_max" value="2000" />		<!-- maximum z velocity in mm/sec = 0.2m/sec -->
	    <param name="control_yaw" value="6.11" />		<!-- maximum rotation rate in radians/sec = 40 degrees per second (1/9 rev/sec) -->

	    <param name="detect_type" value="10" />
	    <param name="detections_select_h" value="32" />	<!-- In the horizontal camera, look for the orange-blue-orange markers -->
	    <param name="detections_select_v_hsync" value="128" />  <!-- In the vertical camera, detect the A4 marker -->
	    <param name="enemy_colors" value="3" />			<!-- Orange-Blue-Orange = 3 -->
	    <param name="enemy_without_shell" value="0" />	<!-- Detect the indoor stickers, rather than the outdoor hull -->
	</node>

	<!-- Launhces multiple Joystick node -->
	<node name="joy_node" pkg="joy" type="joy_node" output="screen" clear_params="true" ns="j0">
		<param name="dev" type="str" value="/dev/input/js0"/>
	</node>
	
	<node name="joy_node" pkg="joy" type="joy_node" output="screen" clear_params="true" ns="j1">
		<param name="dev" type="str" value="/dev/input/js1"/>
	</node>

	<!-- Launches the joystick controller -->
	<node name="joystick_controller" pkg="ardrone_tutorials" type="joystick_controller.py" output="screen" required="true">

		<!-- Namespaces -->
		<param name="NameSpaces"	type="str"	value="[alpha,charlie]" /> <!--no spaces allowed in list-->

		<!-- Configures the joystick button mapping -->
		<param name="ButtonEmergency"	value="8" />
		<param name="ButtonFlattrim"	value="7" />
		<param name="ButtonLand"		value="1" />
		<param name="ButtonTakeoff"		value="0" />

		<!-- Configures the joystick axis mapping -->
		<param name="AxisRoll"			value="0" />
		<param name="AxisPitch"			value="1" />
		<param name="AxisYaw"			value="3" />
		<param name="AxisZ"				value="4" />

		<!-- Configures the joystick mapping -->
		<param name="ScaleRoll"			value="1" />
		<param name="ScalePitch"		value="1" />
		<param name="ScaleYaw"			value="1" />
		<param name="ScaleZ"			value="1" />
	</node>
</launch>
